addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -3053,7 +3053,11 @@ const Wysiwyg = Widget.extend({
         // command is being applied. Note that this needs to be done *before*
         // the command and not after because some commands (e.g. font-size)
         // rely on some elements not to have the class to fully work.
-        for (const node of OdooEditorLib.getSelectedNodes(this.$editable[0])) {
+        const nodes = OdooEditorLib.getSelectedNodes(
+            this.$editable[0], 
+            getDeepRange(this.$editable[0], { splitText: true, select: true })
```

Comment:
I'm afraid calling this on every command might create problems down the line. What is the original issue that you are trying to solve here? That the text node is not returned by `getSelectedNodes` because it is not fully selected?

Reply:
@dmo-odoo That's the case. 
Before the fix :
```xml
<font style="font-size: 62px;" class="o_default_snippet_text">Sell Online. Easily.</font>
```
I select Easily without the dot. And I apply "bold". Result:
```xml
<font style="font-size: 62px;" class="o_default_snippet_text">Sell Online. <strong>Easily</strong>.</font>
```

After the fix :
```xml
<font style="font-size: 62px;">Sell Online. </font>
<font style="font-size: 62px;"><strong>Easily</strong></font>
<font style="font-size: 62px;">.</font>
```

Indeed it's not exactly what I want, but I don't get it. I just want this :

```xml
<font style="font-size: 62px;">Sell Online. <strong>Easily</strong>.</font>
```
Could you help me on this case? It would be great. Thanks

Reply:
@dmo-odoo I think I found a better solution. What do you think?

Reply:
Seeing your second fix, it seems to confirm what I asked you about in my first comment, that is the text node is not returned by `getSelectedNodes` because it is not fully selected. However, your second fix is going to break another case, the one where the selection spans multiple nodes from different containers that have `o_default_snippet_text`. Basically, the original issue is that `getSelectedNodes` has an edge case, that is it won't return a TextNode that is only partially selected. This is working as intended because `getSelectedNodes` is in fact "getFullySelectedNodes". However, that is exactly the case you are trying to solve so it does not help you. That being said, I think `getSelectedNodes` is simply not the right tool for the job. Try with `getTraversedNodes` instead. This one will return any node seen in a DFS traversal between (and including) the start and end of the range, regardless of whether the node is only partially selected or not. I think it is exactly what we want here.

Reply:
I really appreciate the clarification and suggestion to use `getTraversedNodes`. it sounds like the perfect solution. 
I'll go ahead and implement the change based on your feedback. Thank you so much for the valuable insight!